streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -367,7 +367,7 @@ public Subscription subscription(Set<String> topics) {
                         numPartitions = task.partition + 1;
                 }
 
-                internalTopicManager.makeReady(topic, numPartitions);
+                internalTopicManager.makeReady(topic, numPartitions, true);
```

Comment:
Here `topicToTaskIds` contains both internal source topics (should not log compacted) and state changelog topics (log compacted), we should treat them differently.

Actually we have created the internal source topics before triggering the partition grouper, and here it is triggered again in case the number of partitions has changed from the logic of partition grouper.